zkevm-circuits/src/evm_circuit/util/constraint_builder.rs

```diff
@@ -1441,6 +1443,53 @@ impl<'a, F: Field> EVMConstraintBuilder<'a, F> {
         ret
     }
 
+    /// Constraints the next step, given mutually exclusive conditions to determine the next state
+    /// and constrain it using the provided respective constraint. This mechanism is specifically
+    /// used for constraining the internal states for precompile calls. Each precompile call
+    /// expects a different cell layout, but since the next state can be at the most one precompile
+    /// state, we can re-use cells assigned across all those conditions.
+    pub(crate) fn constrain_mutually_exclusive(
+        &mut self,
+        conditions: Vec<Expression<F>>,
+        next_states: Vec<ExecutionState>,
```

Comment:
Very cool. While I consider if it is not need to restrict the usage in `next_step`? i.e. this "mutually_exclusive" feature can be also used in current step. It may make good for some optimizations.

Reply:
In the 'mutually_exclusive' call we have two `cell_manager` objects to maintain: one in `curr` and another in `next`. So in `constraints` fn one can call `constraint_next_step` as he wishes.

Reply:
i will do the rename constrain_mutually_exclusive_next_step and merge